pkg/pillar/zedUpload/datastore_aws.go

```diff
@@ -193,6 +196,30 @@ func (ep *AwsTransportMethod) processS3Download(req *DronaRequest) (error, int)
 	return err, csize
 }
 
+func (ep *AwsTransportMethod) processS3DownloadByChunks(req *DronaRequest) error {
+	pwd := strings.TrimSuffix(ep.apiKey, "\n")
+	dataChan := make(utils.ChunkDataChan)
+	done := make(chan bool)
```

Comment:
I think I understand how "done" is used, but the name is utterly confusing.
This is a way for the producer (the local func here) to wait until the consumer has reached its <-cWriter.Done point i.e., it has passed along the data on the channel. As such the channel could be in the different direction with writeOnChannel writing and this func reading. Is that correct? Basically it is a "proceed to next chunk" signal AFAICT. Is that correct?

But I don't understand how it works when writeOnChannel is called more than once by ProcessChunkByChunk. It will try to read from the channel more than once but this func is only sending once per channel. Please explain.

Reply:
I wanted make the ProcessChunkByChunk to wait until the data is written on to the req.result(writeOnChannel  write into dataNotif and dataNotif will inturn write into req.result ), so that I can make sure all chunks are in order. 

Usually let's say when the file is of size 1.2MB, first 1MB, by the time we write first 1MB into 
final channel, 0.2Mb was getting written.

Reply:
Do you have a test case when ProcessChunkByChunk does multiple calls to writeOnChannel?

Reply:
I don't have a test case, where I have tested manually by uploaded a file of  256kb, 2MB and 6Mb file